unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSRuntime/Components/ECSComponent.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.ECSRuntime
+{
+    public class ECSComponent<ModelType> : IECSComponent
+    {
+        internal Dictionary<string, ECSComponentData<ModelType>> entities = new Dictionary<string, ECSComponentData<ModelType>>();
+        internal Dictionary<string, IECSComponentHandler<ModelType>> handlers = new Dictionary<string, IECSComponentHandler<ModelType>>();
+
+        private readonly Func<IECSComponentHandler<ModelType>> handlerBuilder;
+        private readonly Func<object, ModelType> deserializer;
+        private readonly IParcelScene scene;
+
+        public ECSComponent(IParcelScene scene, Func<object, ModelType> deserializer, Func<IECSComponentHandler<ModelType>> handlerBuilder)
+        {
+            this.scene = scene;
+            this.deserializer = deserializer;
+            this.handlerBuilder = handlerBuilder;
+        }
+
+        /// <summary>
+        /// creates and add component to an entity
+        /// </summary>
+        /// <param name="entity">target entity</param>
+        public void Create(IDCLEntity entity)
+        {
+            var entityId = entity.entityId;
+
+            if (entities.ContainsKey(entityId))
+            {
+                Debug.LogError($"entity {entityId} already contains component {typeof(ModelType)}", entity.gameObject);
+                return;
+            }
+
+            entities[entityId] = new ECSComponentData<ModelType>()
+            {
+                entity = entity,
+                model = default,
+                scene = scene
+            };
+
+            var handler = handlerBuilder?.Invoke();
+
+            if (handler != null)
+            {
+                handlers[entityId] = handler;
+                handler.OnComponentCreated(scene, entity);
+            }
+        }
+
+        /// <summary>
+        /// remove component from entity
+        /// </summary>
+        /// <param name="entity">target entity</param>
+        /// <returns>true if component removed successfully, false if entity didn't contain component</returns>
+        public bool Remove(IDCLEntity entity)
+        {
+            var entityId = entity.entityId;
+            if (handlers.TryGetValue(entityId, out IECSComponentHandler<ModelType> handler))
+            {
+                handler.OnComponentRemoved(scene, entity);
+            }
+            handlers.Remove(entityId);
+            return entities.Remove(entityId);
+        }
```

Comment:
If I am not mistaken, entities live as long as they have components. if a entity doesn't have components, we should remove the entity as well